xonsh/commands_cache.py

```diff
@@ -163,15 +162,18 @@ def lazy_locate_binary(self, name):
                       None)
         if cached:
             return self._cmds_cache[cached][0]
-        elif os.path.isfile(name) and name != os.path.basename(name):
+        elif os.path.isfile(name) and name != pathbasename(name):
             return name
 
     def predict_backgroundable(self, cmd):
         """Predics whether a command list is backgroundable."""
         name = self.cached_name(cmd[0])
-        path, is_alias = self.lazyget(name, (None, None))
-        if path is None or is_alias:
```

Comment:
The first clause is a speed up because calling a function is relatively expensive, the second clause was because this whole formulation doesn't apply to callable aliases at all.